src/main/java/edu/kit/compiler/JavaEasyCompiler.java

```diff
@@ -184,6 +190,12 @@ private static Result compileFirm(String filePath, Logger logger) {
             ast.accept(irv);
             Lower.lower(irv.getTypeMapper());
 
+            for (Optimization optimization : optimizations) {
+                for (Graph graph : Program.getGraphs()) {
+                    optimization.optimize(graph);
```

Comment:
If we swap the order of the two loops and additionally let an optimization return whether anything changed or not, we could do a generic fixpoint iteration that includes all optimizations